### PR TITLE
Rename sorting handlers

### PR DESCRIPTION
**Pull request by lukebjerring**
There are a couple different golang examples, and I much prefer this one (more explicit, simpler to re-use).

### CODE REVIEW

File: metrics/models.go [hunk 1]
```diff
@@ -10,47 +10,27 @@ import (
...
-		return true
-	}
-	if s[i].BrowserName > s[j].BrowserName {
-		return false
+// ByCreatedDate sorts tests by their most-recent created date.
+type ByCreatedDate []base.TestRun
+
+func (s ByCreatedDate) Len() int          { return len(s) }
+func (s ByCreatedDate) Swap(i int, j int) { s[i], s[j] = s[j], s[i] }
+func (s ByCreatedDate) Less(i int, j int) bool {
```
Comment (mdittmer) [inline metrics/models.go]: Maybe add comment that it is descending *Revision*, but ascending *everything else*.

File: metrics/models.go [hunk 2]
```diff
@@ -74,26 +54,18 @@ type TestId struct {
 	Name string `json:"name"`
 }
 
-type TestIdSlice []TestId
-
-func (s TestIdSlice) Len() int {
-	return len(s)
-}
+// ByTestPath sorts test ids by their test path, then name, desc.
```
Comment (mdittmer) [inline metrics/models.go]: nit: Unclear whether `desc` is description or descending.

File: metrics/models.go [hunk 3]
```diff
@@ -10,47 +10,27 @@ import (
...
-	}
-	if s[i].BrowserName < s[j].BrowserName {
-		return true
-	}
-	if s[i].BrowserName > s[j].BrowserName {
-		return false
+// ByCreatedDate sorts tests by their most-recent created date.
+type ByCreatedDate []base.TestRun
+
+func (s ByCreatedDate) Len() int          { return len(s) }
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply metrics/models.go]: Yah, gofmt is checked as part of the linting build.

File: metrics/models.go (hunk 1 above)
Comment (lukebjerring) [reply metrics/models.go]: Done.

File: metrics/models.go (hunk 2 above)
Comment (lukebjerring) [reply metrics/models.go]: Done.

File: metrics/models.go [hunk 4]
```diff
@@ -10,47 +10,27 @@ import (
...
-	}
-	if s[i].Revision > s[j].Revision {
-		return false
-	}
-	if s[i].BrowserName < s[j].BrowserName {
-		return true
-	}
-	if s[i].BrowserName > s[j].BrowserName {
-		return false
+// ByCreatedDate sorts tests by their most-recent created date.
```
[the remark this answers is not in the document]
Comment (lukebjerring) [reply metrics/models.go]: Done.

**Comment by rwaldron** [comment]
How can this be tested/reviewed? Specifically: how can I observe the effect of these changes?

**Comment by lukebjerring** [comment]
Unit tests are how it's tested. You'd have to run the metrics tool in its current state to observe the effects, but I think the state that it's in would only pollute the datastore, until https://github.com/w3c/wptdashboard/pull/334 lands.

**Comment by rwaldron** [comment]
@lukebjerring excellent, that's exactly what I needed to know. Thanks!

**Pull request merged**